Arrowgene.Ddon.Database/Sql/Core/Migration/MailMigration.cs

```diff
@@ -0,0 +1,27 @@
+using System.Data.Common;
+using System.IO;
+using System.Text;
+
+namespace Arrowgene.Ddon.Database.Sql.Core.Migration
+{
+    public class MailMigration : IMigrationStrategy
```

Comment:
Is there an advantage to having a new class file for every migration? It seems like we simply execute sql from the text files. Can we not just read everything directly from the `.sql` itself? Make some set of rules, first line of file has comment which contains the version of the migration or something like that? It seems like other migrators just sort the version numbers, check against the the current version in the table and then just execute in order until it reaches the desired new version. A scheme like this also allows for migrations to be checked in out of order.

Reply:
The purpose is precisely to not be limited to just running SQL files. Being able to run any C# code will allow us to do any kinds of complex behaviour we want, like for example, for the upcoming item migration, which wouldn't be possible, or at least very hard to do with just a SQL script.